scripts/stop-dev.sh

```
@@ -0,0 +1,96 @@
+print_warning() {
+    echo -e "${YELLOW}[WARNING]${NC} $1"
+}
+
+print_error() {
+    echo -e "${RED}[ERROR]${NC} $1"
+}
+
+# Function to kill processes on a specific port
+kill_port() {
+    local port=$1
+    local processes=$(lsof -ti:$port 2>/dev/null || true)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)